simexpal/base.py

```diff
@@ -1328,24 +1362,31 @@ def _update_status_cache_dict(self):
 			if os.access(self.output_file_path('out'), os.F_OK):
 				status, last_mod = Status.STARTED, os.stat(self.output_file_path('out')).st_mtime
 
+
 				jobid = self.slurm_jobid
+				queue_jobid = self.get_queue_jobid()
 				if jobid is not None:
 					if not self._cfg.slurm_queried:
 						self._cfg.query_slurm()
 					status = self._cfg.get_slurm_job_status(jobid)
-
-					return status
+				elif queue_jobid is not None:
+					if not self._cfg.queue_queried:
+						self._cfg.query_queue()
+					status = Status(self._cfg.get_queue_job_status(queue_jobid))
```

Comment:
`self._cfg.get_slurm_job_status` is automatically of type Status-class. Is the same possible (without much hassle) for `self._cfg.get_queue_job_status`?

Reply:
We could type-cast before returning the value in ``get_queue_job_status()``, but I'm not sure if we can do it without type-casting.

Currently we get the job status via communication from the queue socket. The yaml-dumper used in line ``87`` in ``queuesock.py`` can not handle objects. Thats why I originally type-cast the Status to its ``int`` representation and later cast it back to its ``Status`` representation.

Reply:
I type-casted before returning the value in ``get_queue_job_status()`` (and added a comment explaining why it is needed) so the interface is the same for ``get_queue_job_status()`` and ``get_slurm_job_status()``